test/unit/lib/whitehall/asset_manager_storage_test.rb

```diff
@@ -175,23 +176,54 @@ def assets_protected?
       @uploader.store!(@file)
     end
 
-    test "calls worker with model_id and default origin asset_variant" do
-      variant = Asset.variants[:original]
-      asset_args = { assetable_id: @uploader.model.id, asset_variant: variant, assetable_type: @assetable_type }.deep_stringify_keys
+    context "uploader model is AttachmentData" do
+      test "calls worker with assetable_id and default original asset_variant" do
+        variant = Asset.variants[:original]
+        asset_args = { assetable_id: @uploader.model.id, asset_variant: variant, assetable_type: @assetable_type }.deep_stringify_keys
 
-      AssetManagerCreateAssetWorker.expects(:perform_async).with(anything, asset_args, anything, anything, anything, anything)
+        AssetManagerCreateAssetWorker.expects(:perform_async).with(anything, asset_args, anything, anything, anything, anything)
 
-      @uploader.store!(@file)
+        @uploader.store!(@file)
+      end
+
+      test "calls worker with assetable_id and variant/thumbnail asset_variant" do
+        variant = Asset.variants[:thumbnail]
+        @uploader.stubs(:version_name).returns(:thumbnail)
+        asset_args = { assetable_id: @uploader.model.id, asset_variant: variant, assetable_type: @assetable_type }.deep_stringify_keys
+
+        AssetManagerCreateAssetWorker.expects(:perform_async).with(anything, asset_args, anything, anything, anything, anything)
+
+        @uploader.store!(@file)
+      end
     end
 
-    test "calls worker with model_id and variant/thumbnail asset_variant" do
-      variant = Asset.variants[:thumbnail]
-      @uploader.stubs(:store_path).returns("asset-path/thumbnail_file_name")
-      asset_args = { assetable_id: @uploader.model.id, asset_variant: variant, assetable_type: @assetable_type }.deep_stringify_keys
+    context "uploader model is ImageData" do
+      setup do
+        model = build(:image_data)
```

Comment:
not a biggie but this kind of stuff could be written with build(:image_data, use_non_legacy_endpoints: true, id: 1)